cmd2/table_creator.py

```diff
@@ -0,0 +1,748 @@
+                               appearance. Turning off column borders results in a unified appearance between
+                               a row's cells. (Defaults to True)
+        :param bg_odd: optional background color for odd numbered rows (defaults to None)
+        :param bg_even: optional background color for even numbered rows (defaults to gray)
+        """
+        super().__init__(cols, tab_width=tab_width, column_borders=column_borders)
+        self.row_num = 1
+        self.bg_odd = None if bg_odd is None else functools.partial(ansi.style, bg=bg_odd)
+        self.bg_even = None if bg_even is None else functools.partial(ansi.style, bg=bg_even)
+
+    def _apply_bg_color(self, data: Any) -> str:
+        """
+        Convert data to text and apply background color to it based on what row is being generated
+        :param data: data being colored
+        :return converted data
+        """
+        if self.row_num % 2 == 0 and self.bg_even is not None:
+            return self.bg_even(data)
+        elif self.row_num % 2 != 0 and self.bg_odd is not None:
+            return self.bg_odd(data)
+        else:
+            return str(data)
+
+    def generate_data_row(self, row_data: Sequence[Any]) -> str:
+        """
+        Generate a data row
+        :param row_data: Data with an entry for each column in the row.
+        :return: data row string
+        """
+        if self.column_borders:
+            inter_cell = " │ "
+        else:
+            inter_cell = 2 * SPACE
+
+        fill_char = self._apply_bg_color(SPACE)
+        pre_line = self._apply_bg_color("║ ")
+        inter_cell = self._apply_bg_color(inter_cell)
+        post_line = self._apply_bg_color(" ║")
+
+        row = self.generate_row(row_data=row_data, fill_char=fill_char, pre_line=pre_line,
+                                inter_cell=inter_cell, post_line=post_line)
+        self.row_num += 1
+        return row
+
+    def generate_table(self, table_data: Sequence[Sequence[Any]], *, include_header: bool = True) -> str:
+        """
+        Generate a table from a data set
+        :param table_data: Data with an entry for each data row of the table. Each entry should have data for
+                           each column in the row.
+        :param include_header: If True, then a header will be included at top of table. (Defaults to True)
+        """
+        table_buf = io.StringIO()
+
+        if include_header:
+            header = self.generate_header()
+            table_buf.write(header)
+        else:
+            top_border = self.generate_table_top_border()
+            table_buf.write(top_border)
+
+        for index, row_data in enumerate(table_data):
+            # Apply appropriate background color, but don't change the original
+            to_display = list()
+            for col_index, col in enumerate(row_data):
+                to_display.append(self._apply_bg_color(col))
```

Comment:
`index` and `col_index` are unused, and could be replaced with `_` to eliminate pylon errors.

Reply:
This code has been updated